pkg/k8s/user_scoped_client.go

```
@@ -13,14 +13,26 @@ type userScopedClientsets struct {
 	osmV1        *osmv1client.Clientset
 }
 
+// buildUserScopedConfig creates a rest.Config copy that authenticates with the
+// given bearer token instead of any credentials present in the base config.
+// Client certificate fields are cleared because Kubernetes prefers cert auth
+// over bearer tokens when both are present, which would bypass user RBAC.
+func buildUserScopedConfig(baseConfig *rest.Config, userToken string) *rest.Config {
+	cfg := rest.CopyConfig(baseConfig)
+	cfg.BearerToken = userToken
+	cfg.BearerTokenFile = ""
+	cfg.TLSClientConfig.CertData = nil
```

golangci-lint (2.12.2)

[error] QF1008: could remove embedded field "TLSClientConfig" from selector

(staticcheck)